scripts/terraform_schema.py

```diff
@@ -0,0 +1,64 @@
+"""
+Cache Terraform version and schema information in terraform/_schema.json,
+or verify that the cached Terraform information is up-to-date.
+"""
+import argparse
+import json
+import logging
+import os
+import subprocess
+
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+project_root = os.environ['project_root']
+terraform_dir = os.path.join(project_root, 'terraform')
+schema_file = os.path.join(terraform_dir, '_schema.json')
+
+
+def terraform_cmd(*args: str) -> str:
+    cmd = subprocess.run(['terraform', *args],
+                         cwd=terraform_dir,
+                         check=True,
+                         stdout=subprocess.PIPE,
+                         text=True,
+                         shell=False)
+    return cmd.stdout
+
+
+def generate_schema_json() -> None:
+    schema = terraform_cmd('providers', 'schema', '-json')
+    versions, footer = terraform_cmd('-version').split('\n\n')
+    with open(schema_file, 'w') as f:
+        json.dump({
+            'versions': versions,
+            'schema': json.loads(schema)
+        }, f)
+
+
+def check_schema_json() -> None:
+    versions, footer = terraform_cmd('-version').split('\n\n')
+    with open(schema_file, 'r') as f:
+        versions_cached = json.load(f)['versions']
```

Comment:
This isn't exactly caching. A cache usually isn't perturbed by being deleted.

Reply:
Hannes prefers "tracked" or "committed" in that order